src/workerd/io/actor-cache.c++

```diff
@@ -2732,18 +2733,33 @@ kj::Promise<void> ActorCache::flushImplUsingTxn(
   auto promises = kj::heapArrayBuilder<kj::Promise<void>>(
       rpcPuts.size() + rpcMutedDeletes.size() + rpcCountedDeletes.size()
       + 2 + !maybeAlarmChange.is<CleanAlarm>());
+
+  auto joinCountedDelete = [](RpcCountedDelete& rpcCountedDelete) -> kj::Promise<void> {
+    auto promises = KJ_MAP(request, rpcCountedDelete.rpcDeletes) {
+      return request.send().then(
+          [](capnp::Response<rpc::ActorStorage::Operations::DeleteResults>&& response) mutable
+          -> uint {
+        return response.getNumDeleted();
+      });
+    };
+
+    for (auto& promise : promises) {
+      // Reuse `countDeleted` since it's already in a state object anyway.
+      rpcCountedDelete.countedDelete->countDeleted += co_await promise;
+    }
+  };
```

Comment:
Ha, I see what you mean now about all of this being a build up just to get the accurate delete count. Oof.